homeassistant/components/august/__init__.py

```diff
@@ -170,6 +181,39 @@ def close_http_session(event):
     return setup_august(hass, config, api, authenticator)
 
 
+class AugustHTTPSession(Session):
+    """Wrap a requests Session object so we can retry on http error 429."""
+
+    # This resolves the 429 issue while
+    # waiting for upstream to process https://github.com/snjoetw/py-august/pull/24
+
+    def request(self, method, url, **kwargs):
+        """Wrap the request in order to adjust the User-Agent and retry on 429."""
+        attempts = 0
+        response = None
+
+        # We get a higher rate limit if we pass a newer
+        # version so lets try to avoid the 429 up front
+        kwargs["headers"][
+            "User-Agent"
+        ] = "August/2019.12.16.4708 CFNetwork/1121.2.2 Darwin/19.3.0"
```

Comment:
This should not be in Home Assistant. This should be in the lib.

Reply:
We won't accept this code in home assistant. If the library maintainer is unresponsive for a reasonable time, eg more than one month, and theres no other activity in the library, the library can be forked.

Please try to contact the maintainer to establish a plan going forward. Maybe the maintainer is willing to transfer ownership of the library?

Reply:
@MartinHjelmare The py-august maintainer has responded and released a new version.  PR resubmitted. 

This should resolve @balloob 's review comments